cloud/team/team.go

```diff
@@ -757,3 +758,202 @@ func ListTeamUsers(teamID string, out io.Writer, client astrocore.CoreClient) (e
 	fmt.Println("The selected team has no members")
 	return nil
 }
+
+func GetDeploymentTeams(client astrocore.CoreClient, deployment string, limit int) ([]astrocore.Team, error) {
+	offset := 0
+	var teams []astrocore.Team
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		includeDeploymentRoles := true
+		resp, err := client.ListDeploymentTeamsWithResponse(httpContext.Background(), ctx.Organization, deployment, &astrocore.ListDeploymentTeamsParams{
+			IncludeDeploymentRoles: &includeDeploymentRoles,
+			Offset:                 &offset,
+			Limit:                  &limit,
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		teams = append(teams, resp.JSON200.Teams...)
+
+		if resp.JSON200.TotalCount <= offset {
+			break
+		}
+
+		offset += limit
+	}
+
+	return teams, nil
+}
+
+// Prints a list of all of an organizations teams
+func ListDeploymentTeams(out io.Writer, client astrocore.CoreClient, deployment string) error {
+	table := printutil.Table{
+		DynamicPadding: true,
+		Header:         []string{"ID", "Role", "Name", "Description", "Create Date"},
+	}
+	teams, err := GetDeploymentTeams(client, deployment, teamPagnationLimit)
+	if err != nil {
+		return err
+	}
+
+	for i := range teams {
+		var teamRole string
+		for _, role := range *teams[i].Roles {
+			if role.EntityType == "DEPLOYMENT" && role.EntityId == deployment {
+				teamRole = role.Role
+			}
+		}
+		table.AddRow([]string{
+			teams[i].Id,
+			teamRole,
+			teams[i].Name,
+			*teams[i].Description,
```

Comment:
do we have to ever do a nil check here or should it be safe without?

Reply:
done. also added to the other three locations we were just assuming this was not nil